bcc/table.go

```diff
@@ -180,40 +144,40 @@ func (table *Table) Iter() <-chan Entry {
 		alternateKeys := []byte{0xff, 0x55}
 		res := C.bpf_lookup_elem(fd, keyP, leafP)
 		// make sure the start iterator is an invalid key
-		for i := 0; i <= len(alternateKeys); i++ {
+		for i := 0; i < len(alternateKeys); i++ {
 			if res < 0 {
 				break
 			}
 			for j := range key {
+				fmt.Println(len(key), len(alternateKeys))
 				key[j] = alternateKeys[i]
 			}
-			res = C.bpf_lookup_elem(fd, keyP, leafP)
 		}
 		if res == 0 {
 			return
 		}
-		keyStr := make([]byte, key_size*8)
-		leafStr := make([]byte, leaf_size*8)
-		keyStrP := (*C.char)(unsafe.Pointer(&keyStr[0]))
-		leafStrP := (*C.char)(unsafe.Pointer(&leafStr[0]))
 		for res = C.bpf_get_next_key(fd, keyP, keyP); res == 0; res = C.bpf_get_next_key(fd, keyP, keyP) {
```

Comment:
bcc has a function [`bpf_get_first_key`](https://github.com/iovisor/bcc/blob/fdf9b08cee7844d9ad86cbb1ab6b5adeff2b0520/src/cc/libbpf.c#L141) now, so you could also remove the initialization with an invalid key above and reduce this to

```
for res = C.bpf_get_first_key(fd, keyP, keySize); res == 0; res = C.bpf_get_next_key(fd, keyP, keyP) {
```